SpatialGDK/Source/SpatialGDKTests/Public/GDKAutomationTestBase.h

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+#include "Misc/AutomationTest.h"
+#include "SpatialGDKTests/SpatialGDKServices/LocalDeploymentManager/LocalDeploymentManagerUtilities.h"
+
+DEFINE_LOG_CATEGORY_STATIC(LogGDKTestBase, Log, All);
+
+/**
+ * This class extends the Unreal AutomationTestBase to allow unit tests to be augmented with a set-up step, called
+ * before each test.
+ * This class is then offered through a macro, in a similar way to `IMPLEMENT_SIMPLE_AUTOMATION_TEST`.
+ *
+ * To use this test base, the GDK_AUTOMATION_TEST macro should be used, followed by the test body:
+ * ```
+ *	   GDK_AUTOMATION_TEST(MyModule, MyComponent, MyTestName)
+ *	   {
+ *			// do some testing here...
+ *
+ *			return true;
+ *	   }
+ * ```
+ *
+ * Returning `true` indicates a test pass and returning `false` indicates test failure.
+ */
+class FGDKAutomationTestBase : public FAutomationTestBase
+{
+public:
+	FGDKAutomationTestBase(const FString& InName, bool bInComplexTask)
+		: FAutomationTestBase(InName, bInComplexTask)
+	{
+	}
+};
+
+#define GDK_AUTOMATION_TEST(ModuleName, ComponentName, TestName)                                                                           \
+	IMPLEMENT_GDK_AUTOMATION_TEST(TestName, "SpatialGDK." #ModuleName "." #ComponentName "." #TestName)
+
+#define IMPLEMENT_GDK_AUTOMATION_TEST(TestName, PrettyName)                                                                                \
```

Comment:
Can this be templated instead of macro'd?

Reply:
Or maybe most of the code implemented in the base using CRTP (curiously recurring template pattern)?

Reply:
Code in macros scares me and is hard to debug

Reply:
I don't think macros are avoidable since each test is a class and `__FILE__` and `__LINE__` are required for useful test logging

Miron's suggestion of reducing the macro code as much as possible might help this concern, though (pushing now)

Reply:
Ah I see, you could pass those in as arguments and return from the virtual methods